library/Freckle/App/Prelude.hs

```diff
@@ -0,0 +1,137 @@
+-- | Those functions and types we can't do without!
+module Freckle.App.Prelude
+  (
+  -- * 'Prelude' as the starting point, without common partial functions
+    module Prelude
+
+  -- * Commonly imported types
+  , Alternative
+  , Exception
+  , Generic
+  , HashMap
+  , Hashable
+  , Int64
+  , Map
+  , MonadIO
+  , MonadReader
+  , MonadUnliftIO
+  , NominalDiffTime
+  , NonEmpty
+  , PrimMonad
+  , ReaderT
+  , Set
+  , Text
+  , UTCTime
+  , Vector
+
+  -- * Commonly used functions
+
+  -- ** Lifts
+  , lift
+  , liftIO
+
+  -- ** 'Text'
+  , tshow
+
+  -- ** 'Maybe'
+  , catMaybes
+  , fromMaybe
+  , isJust
+  , isNothing
+  , listToMaybe
+  , mapMaybe
+  , maybeToList
+  , readMay
+
+  -- ** 'Either'
+  , partitionEithers
+
+  -- ** 'Foldable'
+  , fold
+  , foldlM
+  , for_
+  , toList
+  , traverse_
+
+  -- ** 'Traversable'
+  , for
+
+  -- ** 'Functor'
+  , (<$$>)
+
+  -- ** 'Bifunctor'
+  , bimap
+  , first
+  , second
+
+  -- ** 'Applicative'
+  , (<|>)
+  , liftA2
+  , optional
+
+  -- ** 'Monad'
+  , (<=<)
+  , (>=>)
+  , guard
+  , join
+  , unless
+  , void
+  , when
+
+  -- ** 'Arrow'
+  , (&&&)
+  , (***)
+
+  -- ** Function
+  , on
+
+  -- ** 'UTCTime'
+  , getCurrentTime
+  ) where
+
+-- Use 'Prelude' as the starting point, removing common partial functions
+
+import Prelude hiding
+  (cycle, foldl1, foldr1, head, init, last, maximum, minimum, read, tail, (!!))
```

Comment:
I love that you are hiding some partial functions. Can we export their friends from `Safe` (`headMay`, `readMay`, etc) and `semigroupoids` (`fold1`, `foldMap1`, etc).